### ISSUE TITLE
Cerberus not working properly - Interrupting idle Thread

### ISSUE THREAD

**Issue by mbessa** [cerberustesting/cerberus-source#1202]
Hello,

As discussed this week, when Cerberus stop working correctly we start getting lot of similar errors in glassfish logs. "Interrupting idle Thread....._cerberus instruction_" error seems to be directly related with test PE executions.

This have been a recurrent issue and could be related with network issues/high server load.
Unless we are looking at the glassfish logs, these issues only start to being noticed when most of the tests that use queries to DB's start failing.

Since the recovery for this issue involves restart(or kill) clusters/instances and clear hub's/nodes from pending executions for all nodes, it's time consuming and unfortunately has been happening many times a day.
Most of the times we can't figure out what trigger these errors, but when start generating these errors, spreads through all Cerberus clusters/instances. After the recovery is done, server could be running fine for a few hours without problems or only a few minutes. Many times we weren't even running campaigns that could cause high server load.

Can you please investigate this? It is really urgent for us that we can have Cerberus stable. At this point we have high impact in Regression campaigns and in Performance Monitoring.

Logs:
```
==> instanceCerberusMonitor02/logs/server.log <==
[#|2017-02-16T11:18:42.321+0100|WARNING|glassfish3.1.2|com.sun.grizzly.config.GrizzlyServiceListener|_ThreadID=52;_ThreadName=Thread-2;|GRIZZLY0023: Interrupting idle Thread: Forwarding findElement on session ab46561d-1e26-45c8-89f9-26b9437d2da2 to remote.|#]

[#|2017-02-16T11:18:42.322+0100|WARNING|glassfish3.1.2|com.sun.grizzly.config.GrizzlyServiceListener|_ThreadID=52;_ThreadName=Thread-2;|GRIZZLY0023: Interrupting idle Thread: Forwarding isElementDisplayed on session 4ed519b4-78e4-4684-9f56-c8945dec06c8 to remote.|#]

[#|2017-02-16T11:18:42.322+0100|WARNING|glassfish3.1.2|com.sun.grizzly.config.GrizzlyServiceListener|_ThreadID=52;_ThreadName=Thread-2;|GRIZZLY0023: Interrupting idle Thread: Forwarding executeScript on session 4c3e7d12-8f33-4082-89b9-59752834d8b2 to remote.|#]


==> instanceCerberusMonitor01/logs/server.log <==
[#|2017-02-16T11:18:42.473+0100|WARNING|glassfish3.1.2|com.sun.grizzly.config.GrizzlyServiceListener|_ThreadID=51;_ThreadName=Thread-2;|GRIZZLY0023: Interrupting idle Thread: Forwarding executeScript on session ae10ad34-cbff-4a28-8211-2c649a5eefa0 to remote.|#]

[#|2017-02-16T11:18:42.474+0100|WARNING|glassfish3.1.2|com.sun.grizzly.config.GrizzlyServiceListener|_ThreadID=51;_ThreadName=Thread-2;|GRIZZLY0023: Interrupting idle Thread: Forwarding findElement on session 9a1e7f19-2d85-45db-9d99-d98284178d1f to remote.|#]


==> instanceCerberusMonitor03/logs/server.log <==
[#|2017-02-16T11:18:42.526+0100|WARNING|glassfish3.1.2|com.sun.grizzly.config.GrizzlyServiceListener|_ThreadID=53;_ThreadName=Thread-2;|GRIZZLY0023: Interrupting idle Thread: Forwarding isElementDisplayed on session 91778a5e-9111-4361-95ca-efe101f3f2bc to remote.|#]

[#|2017-02-16T11:18:42.527+0100|WARNING|glassfish3.1.2|com.sun.grizzly.config.GrizzlyServiceListener|_ThreadID=53;_ThreadName=Thread-2;|GRIZZLY0023: Interrupting idle Thread: Forwarding findElement on session 67030efb-0fe2-4a69-b3ed-2af49d4e6b12 to remote.|#]
```

Thanks in advance for your help,
Marco

Cerberus Version : 1.1.11-SNAPSHOT-1084

**Comment by abourdon** [comment]
Hi @mbessa,

When you said Cerberus stop working properly, did you mean execution campaign is blocked? How do you launch campaign execution: thanks GUI by using the run multiple page or by using scripts which make calls to Cerberus API?

Regards,
Aurélien

**Comment by mbessa** [comment]
Hello Aurélien,

Performance tests by script using cron (as always). Regression also by script, triggered manually, but there were no changes in the way these tests were launched.
If tests are executed directly from Cerberus the results are the same. Can't gather data from DB.

Lately this problem seems to be more frequent due to network issues (glitches at DC) and/or high server load and/or Front's response.
Basically, every time the conditions are not ideal, Cerberus lose connection with running tests leaving them "hanging" PE for hours (of course this effect is magnified when you run campaigns + performance tests and other campaigns/tests by other teams), after pile up some PE tests, the one's that use connections to DB stop gather data from DB's, other tests also take longer to complete or fail randomly, sometimes GL instances stop working and have to be killed.
The only pattern we found are these messages in logs.

Any idea ?
BR
Marco

**Comment by abourdon** [comment]
Not really but we have to investigate it quickly.

Would you have an idea of ​​the date of the first problems?

**Comment by mbessa** [comment]
These problems have been more present in the last two weeks, before happened from time to time.

**Comment by mbessa** [comment]
Do you have any news on this subject? So far the problem persists.

**Comment by bcivel** [comment]
Hi @mbessa .
fixed with that commit 581bd12
Thanks
Benoit
